packages/teleport/src/Discover/SelectResource.tsx

```diff
@@ -0,0 +1,245 @@
+import styled from 'styled-components';
+
+import { Image, Text, Box, ButtonPrimary, ButtonSecondary, Flex } from 'design';
+
+import Empty from 'teleport/components/Empty';
+import cfg from 'teleport/config';
+
+import { resourceTypes } from './resource-lists';
+
+import applicationIcon from './assets/application.png';
+import databaseIcon from './assets/database.png';
+import serverIcon from './assets/server.png';
+import k8sIcon from './assets/kubernetes.png';
+
+import type { TabComponent } from 'design/SlideTabs/SlideTabs';
+import type { ResourceType, ResourceLocation } from './resource-lists';
+
+export function SelectResource({ onSelect }: SelectResourceProps) {
+  const [selectedResource, setSelectedResource] = useState<string>('server');
+  const [selectedType, setSelectedType] = useState('');
+  const [disableProceed, setDisableProceed] = useState<boolean>(true);
+  const tabs: TabComponent[] = [
+    {
+      name: 'server',
+      component: (
+        <Flex style={{ lineHeight: '31px' }}>
+          <Image src={serverIcon} width="32px" mr={2} /> Server
+        </Flex>
+      ),
+    },
+    {
+      name: 'database',
+      component: (
+        <>
+          <Flex style={{ lineHeight: '31px' }}>
+            <Image src={databaseIcon} width="32px" mr={2} /> Database
+          </Flex>
+        </>
+      ),
+    },
+
+    {
+      name: 'kubernetes',
+      component: (
+        <Flex style={{ lineHeight: '31px' }}>
+          <Image src={k8sIcon} width="32px" mr={2} /> Kubernetes
+        </Flex>
+      ),
+    },
+
+    {
+      name: 'application',
+      component: (
+        <Flex style={{ lineHeight: '31px' }}>
+          <Image src={applicationIcon} width="32px" mr={2} /> Application
+        </Flex>
+      ),
+    },
+
+    {
+      name: 'desktop',
+      component: (
+        <Flex style={{ lineHeight: '31px' }}>
+          <Image src={serverIcon} width="32px" mr={2} /> Desktop
+        </Flex>
+      ),
+    },
+  ];
+
+  useEffect(() => {
+    if (selectedResource === 'server') {
+      // server doesn't have any additional deployment options
+      setDisableProceed(false);
+      return;
+    }
+    if (selectedResource && selectedType) {
+      setDisableProceed(false);
+      return;
+    }
+    setDisableProceed(true);
+  }, [selectedResource, selectedType]);
+
+  return (
+    <Box width="1020px">
+      <Text mb={4} typography="h4">
+        Resource Selection
+      </Text>
+      <Text mb={2}>Select Resource Type</Text>
+      <SlideTabs
+        tabs={tabs}
+        onChange={index => setSelectedResource(tabs[index].name)}
+      />
+      {selectedResource === 'database' && (
+        <SelectDBDeploymentType
+          selectedType={selectedType}
+          setSelectedType={setSelectedType}
+          resourceTypes={resourceTypes}
+        />
+      )}
+      {selectedResource === 'application' && (
+        <Empty
+          clusterId="abc123"
+          canCreate={true}
```

Comment:
i think this flag should come from perms defined in user context?

Reply:
Oh yeah 100% this doesn't work at all, it was just put here as a placeholder to see how it would work. :)